src/Refactor/EntityWriter/ProductXmlWriter.php

```diff
@@ -0,0 +1,238 @@
+<?php
+namespace DemandwareXml\Refactor\EntityWriter;
+
+use DemandwareXml\Refactor\Entity\Product;
+use DemandwareXml\Refactor\Xml\{XmlFormatter, XmlWriter};
+
+class ProductXmlWriter
+{
+    private $writer;
+    private $product;
+
+    public function __construct(XmlWriter $writer, Product $product)
+    {
+        $this->writer  = $writer;
+        $this->product = $product;
+    }
+
+    public function write(): void
+    {
+        $this->writer->startElement('product');
```

Comment:
One nice thing about this refactor is that the order things are written is the order they appear in the XML. In the existing codebase that's not the case, and so the only way to work it out is to look at a real world file and hope it contains all the elements.